modules/ting_das/includes/ting_das.client.inc

```diff
@@ -0,0 +1,431 @@
+   *   Article year.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setPublicationYearOfComponent(string $publicationYearOfComponent) {
+    $this->publicationYearOfComponent = $publicationYearOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get the article year.
+   *
+   * @return string|null
+   *   The article year.
+   */
+  public function getPublicationYearOfComponent() {
+    return $this->publicationYearOfComponent;
+  }
+
+  /**
+   * Set article issue number.
+   *
+   * @param string $issueOfComponent
+   *   Article issue number.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setIssueOfComponent(string $issueOfComponent) {
+    $this->issueOfComponent = $issueOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get article issue number.
+   *
+   * @return string
+   *    The issue number.
+   */
+  public function getIssueOfComponent() {
+    return $this->issueOfComponent;
+  }
+
+  /**
+   * Set the the volume number.
+   *
+   * @param string $volumeOfComponent
+   *   The article volume number.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setVolumeOfComponent(string $volumeOfComponent) {
+    $this->volumeOfComponent = $volumeOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get the article volume number.
+   *
+   * @return string|null
+   *   The article volume number.
+   */
+  public function getVolumeOfComponent() {
+    return $this->volumeOfComponent;
+  }
+
+  /**
+   * Set the page number in the publication.
+   *
+   * @param int $pagesOfComponent
+   *   The page number.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setPagesOfComponent(int $pagesOfComponent) {
+    $this->pagesOfComponent = $pagesOfComponent;
+    return $this;
+  }
+
+  /**
+   * Get the page number.
+   *
+   * @return string|null
+   *   The page number.
+   */
+  public function getPagesOfComponent() {
+    return $this->pagesOfComponent;
+  }
+
+  /**
+   * Send request to the service.
+   *
+   * @return bool
+   *   TRUE if request was send else FALSE.
+   */
+  public function send() {
+    $options = array(
+      'headers' => array(
+        'Content-Type' => 'application/xml; charset=UTF8',
+      ),
+      'body' => $this->buildMessage(),
+    );
+
+    try {
+      $this->client->request('POST', $this->url, $options);
+
+      // We don't look at the response but assumes that everything is okay. This
+      // is do to the fact that the service at current time only returns 204
+      // regardless of what was sent as long as the XML message is correct. If
+      // the message is wrong a non 200 code will be returned, which will
+      // trigger and guzzle exception (handled below).
+      return TRUE;
+    }
+    catch (GuzzleException $exception) {
+      watchdog_exception('ting_das', $exception);
```

Comment:
This code has been updated but for some reason GitHub will not let me continue the discussion of this. Anyhow:

Not having Drupal specific code in our libraries is a general concern. For logging it would be useful across all our clients. Adding a library for handling this will have to start somewhere. It might as well be here.

If we do not want to add the library then we could at least just not do error handling internally and just let `send` throw an exception. Then the calling Drupal code could catch and call `watchdog_exception()` accordingly.

Reply:
fixed